parser/src/tests/errors.rs

```diff
@@ -0,0 +1,72 @@
+use crate::error::ParseError::InvalidInt;
+use crate::{build_parse_test, error::Error, tests::ParseTest};
+
+// SCAN ERRORS
+// ================================================================================================
+
+#[test]
+fn error_invalid_token() {
+    let source = "
+    transition_constraints:
+        enf clk'' = clk + 1";
+    // Only one "'" is allowed at the end of identifier to indicate next row of the column.
+    // Should fail if more than one "'" is present in an identifier
+    let error = Error::ScanError(45..46);
+    build_parse_test!(source).expect_error(error);
+}
+
+#[test]
+fn error_identifier_with_invalid_characters() {
+    let source = "
+    transition_constraints:
+        enf clk@' = clk@ + 1";
+    // "@" is not in the allowed characters.
+    let expected = Error::ScanError(44..45);
+    build_parse_test!(source).expect_error(expected);
+}
+
+#[test]
+fn error_invalid_symbol() {
+    let source = "
+    transition_constraints:
+        enf clk' = clk / 1";
+    // "/" is not a valid token.
+    let expected = Error::ScanError(52..53);
+    build_parse_test!(source).expect_error(expected);
+}
+
+// PARSE ERRORS
+// ================================================================================================
+
+#[test]
+fn error_invalid_int() {
+    let num: i128 = i64::max_value() as i128 + 1;
+    let source = format!(
+        "
+    transition_constraints:
+        enf clk' = clk + {}",
+        num.to_string()
+    );
+    // Integers can only be of type i64.
+    let error = Error::ParseError(InvalidInt(format!("Int too big : {}", num.to_string())));
+    build_parse_test!(source.as_str()).expect_error(error);
+}
+
+// UNRECOGNIZED TOKEN ERRORS
+// ================================================================================================
+
+#[test]
+fn error_constraint_without_section() {
+    // Constraints outside of valid sections are not allowed.
+    let source = "enf clk' = clk + 1";
+    build_parse_test!(source).expect_unrecognized_token();
+}
+
+#[test]
+fn error_identifier_starting_with_int() {
+    // Identifiers cannot start with numeric characters.
+    let source = "
+    transition_constraints:
+        enf 1clk' = 1clk + 1";
```

Comment:
shouldn't this be a `ScanError` rather than failing at the parser level?

Reply:
Thanks for the catch. Logos tokenizes this into `Number(1)` and `Ident(clk)` so this doesn't fail at the lexer level. I've added an explanation for this test and added a few other tests to cover some potential confusing cases.

Reply:
Great, thank you!